### ISSUE TITLE
Consider adding binder links to jupyter example notebooks

### ISSUE THREAD

**Issue by djhoese** [vispy/vispy#2174]
sphinx-gallery allows you to create binder links for examples:

https://sphinx-gallery.github.io/stable/configuration.html?highlight=notebook#generate-binder-links-for-gallery-notebooks-experimental

Pro: The jupyter_rfb examples could be easily run by users.

Con: Some examples won't work (specifically the Qt GUI ones) and I'm not sure if/how we can distinguish (turn off) which examples should get the binder link.

Thoughts @almarklein and @larsoner ?

**Comment by larsoner** [comment]
One option would be to not execute the Qt examples during the doc build. You could for example add a `_noplot` to the end like matplotlib does, and set the `filename_pattern` accordingly

**Comment by djhoese** [comment]
Yeah, but I actually have the scraper working for Qt GUIs which would be too bad if they couldn't be shown. I want to have my cake and eat it too please.

**Comment by almarklein** [comment]
We already add metadata  (e.g. `# vispy: gallery 2`) to the examples, Can we extend that, with e.g. `# vispy: gallery 2 binder`? Or is this metadata consumed by Sphinx-gallery without a way for us to intervene?

**Comment by almarklein** [comment]
>  Some examples won't work (specifically the Qt GUI ones) 

If we make the examples select a backend automatically, it should just work, right? What proportion would not work? We could add a comment in these that these need to be run on desktop, perhaps.

**Comment by djhoese** [comment]
> What proportion would not work?

There are a couple examples that are larger Qt GUIs with other Qt widgets (buttons, sliders, dropdowns, etc) and not just a VisPy canvas.

**Comment by almarklein** [comment]
Well, if only a few, then a comment should suffice, IMO.

**Comment by djhoese** [comment]
Yeah, true. I'm ok with that.

**Comment by almarklein** [comment]
BTW: If we are to add binder links, we should add a `binder/environment.yml` (like [this](https://github.com/vispy/jupyter_rfb/tree/main/binder) to help Binder setup a working environment.

**Comment by djhoese** [comment]
Given that the binder environment is rebuilt on every commit (and I'm not sure there is a way to customize that) maybe we should make a new repository that is just the binder environment; like `github.com/vispy/vispy-binder-env`. Then our binder links could point to that repository for the environment and this main repository for the notebooks.

**Comment by almarklein** [comment]
That makes a lot of sense. The only (minor) downside is that we should bump something in that repo on each release.

**Comment by djhoese** [comment]
Another thing I think I've seen is people create GitHub workflows for triggering the rebuild of the binder image. We could probably do that for either setup (single or two repos).
